pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1003,38 +1003,172 @@ func TestHeartbeatCancel(t *testing.T) {
 	require.ErrorContains(t, <-retryErrorChan, fmt.Sprintf("replication stream %s is not running, status is STREAM_INACTIVE", prodAID))
 }
 
-func setupLogicalTestServer(
-	t *testing.T, ctx context.Context, clusterArgs base.TestClusterArgs, numNodes int,
-) (
-	*testcluster.TestCluster,
-	serverutils.ApplicationLayerInterface,
-	*sqlutils.SQLRunner,
-	*sqlutils.SQLRunner,
-) {
+func TestAcyclicReplication(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "multi cluster/node config exhausts hardware")
+
+	ctx := context.Background()
+
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestControlsTenantsExplicitly,
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+				DistSQL:          &execinfra.TestingKnobs{},
+			},
+		},
+	}
+
+	numDBs := 3
+	server, s, runners := setupServerWithNumDBs(t, ctx, clusterArgs, 1, numDBs)
+	defer server.Stopper().Stop(ctx)
+
+	PGURLs, cleanup := GetPGURLs(t, s, numDBs)
+	defer cleanup()
+
+	jobIDs := make([]jobspb.JobID, 3)
+	for i := range numDBs {
+		// Set a dummy jobID to make the array logic simpler
+		if i == 0 {
+			jobIDs[0] = jobspb.InvalidJobID
+			continue
+		}
+		runners[2].QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", PGURLs[i].String()).Scan(&jobIDs[i])
+	}
+
+	// Insert into C, then check B -> C wins
+	runners[2].Exec(t, "UPSERT INTO tab VALUES (1, 'hello')")
```

Comment:
i think the main thing we want to test is that an update to a0 does not propogate to a2. I don't see how the current writes in this test relate specifically to the behavior in acyclic replication.

Reply:
I'm actually aiming to test A,B -> C works as intended, based on [this doc](https://cockroachlabs.atlassian.net/wiki/spaces/BI/pages/edit-v2/3856695328?draftShareId=96d0b89e-b40c-44f4-bd40-017f676cb5cc) in the section where I discuss A -> B -> C not working. This is technically acyclic but I can rename it if needed.

Reply:
Changed the name

Reply:
Oh whoops. yes.